feat. history file improvements

Rendevior opened the pull request:
- minor changes

Rendevior committed:
feat. deleting a specific entry in history

- minor changes

Rendevior committed:
docs: changes

Rendevior committed:
feat. history update

> for updating history entries if theres new episodes released
+ style: change prompt

Rendevior committed:
chore: shellcheck

Rendevior committed:
docs: minor changes

- added syntax highlighting (sh)

TheGlockMisc commented:
soon will update the preview video. :)

Rendevior committed:
refactor: GNU-ism

TheGlockMisc commented:
found out that `sed -i` isn't a POSIX, so we're going back to good ol' iterating/slow loops, instead of using shell jobs to iterate simultainiously. though, i'm still gonna improve it a little.

Rendevior committed:
fix-up: multi-thread in updating history

+ some enhancements
+ removed quotes with exit code number
+ added timeouts in scraping

TheGlockMisc commented:
> found out that `sed -i` isn't a POSIX, so we're going back to good ol' iterating/slow loops, instead of using shell jobs to iterate simultainiously. though, i'm still gonna improve it a little.

Aight, i knew it.. there was a POSIX-way workaround of `sed -i`
I'm done with this one.

Rendevior commented on hayanime:
```diff
@@ -232,8 +232,8 @@ ffmpegdler(){
-		[ -w ~/.hayanihist ] || die "1" "Hist file is Unwritable, Make sure you have access to write to HOME directory"
-		sed "/${3}/d" ~/.hayanihist > ~/.hayanihist.tmp && mv ~/.hayanihist.tmp ~/.hayanihist
+		[ -w ~/.hayanihist ] || die 1 "Hist file is Unwritable, Make sure you have access to write to HOME directory"
+		printf '%s' "$(sed "/${3}/d" ~/.hayanihist)" > ~/.hayanihist
```
Note that it expands the command substitution first before writing/redirecting it directly to the same file

Rendevior commented on hayanime:
```diff
@@ -332,14 +332,22 @@ hist_handler(){
 		unset cur_ep
 	done < ~/.hayanihist
+	wait "${!}"
+	[ -z "$(cat ~/.hayanihist.tmp)" ] && die 1 "\033[2K\rFailed to Update..."
```
If it fails it will not write the update to the main history file

Rendevior committed:
fix: causing error even it succeed

fearocanity commented:
Zamn, all working fine and already tested it. this version would be my primary one.
